user groups: Minor fixes in CSS common to #groups and #streams overlay.

m-e-l-u-h-a-n opened the pull request:
This PR does following improvements in #streams and #groups overlay.
* Adds word break in the title in the right pane to handle long stream and group names.
* Handle long group/stream names in group/stream name edit dialog box header.
* Fix positioning of edit (pencil) button to the right end and provide maximum available space for stream/group name.

Follow-up for: #24514 (and edit for #24583, originally worked by @Ddharmani3).

<!-- Describe your pull request here.-->
<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**
Long stream name title on right pane:
![image](https://user-images.githubusercontent.com/63504956/226007881-40b4ee64-068a-4d55-948b-6ef787c292c4.png)
Long group name title in right pane:
![image](https://user-images.githubusercontent.com/63504956/226009004-77ccc3e8-d4d2-4935-80c1-7774169a8a7e.png)

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

m-e-l-u-h-a-n commented:
@alya I have reviewed #24583 and included edits in this commit. However I am not sure about how we should keep long stream/group names in stream/group name edit dialog box.
@Ddharmani3 had kept the full names with word breaks in #24583 .
![Keeping-full-names-with-word-breaks](https://user-images.githubusercontent.com/63504956/226015315-4fe5c778-0590-4386-aa05-20272c745a23.png)

and I have put ellipsis circles to the overflowing group/stream names.
![ellipsis-to-oveorflowing-names](https://user-images.githubusercontent.com/63504956/226016793-815b9f7c-4f3e-431c-a8a0-04869293b378.png)


It would be helpful if you could suggest which one we should go with and I would update the pr accordingly.

Ddharmani3 commented:
> @alya I have reviewed #24583 and included edits in this commit. However I am not sure about how we should keep long stream/group names in stream/group name edit dialog box. @Ddharmani3 had kept the full names with word breaks in #24583 . 
> 
> and I have put ellipsis circles to the overflowing group/stream names.
> 
> It would be helpful if you could suggest which one we should go with and I would update the pr accordingly.
I had originally used the full group name, thinking that the user would be able to see the full name while editing. However, the ellipsis look better from a UI perspective.

Is it possible to add a tooltip that shows the full name on hover when using the abbreviated name?

sahil839 commented on web/styles/subscriptions.css:
```diff
@@ -787,25 +794,25 @@ h4.user_group_setting_subsection_title {
 
         .group-name-wrapper,
         .stream-name {
-            display: inline-block;
```
Why this has been removed? I guess it should not have any effect. And if we want to remove it for being redundant then it can be done in separate commit.

sahil839 reviewed (commented):
Posted a couple of comments.

One thing about stream settings title - I think we would want the stream symbol to be on the same line as name. 

![Screenshot from 2023-03-21 13-26-04](https://user-images.githubusercontent.com/35494118/226546787-74931c1b-5a2a-49cc-87f8-ead69644edd7.png)

m-e-l-u-h-a-n commented on web/styles/subscriptions.css:
```diff
@@ -843,7 +850,8 @@ h4.user_group_setting_subsection_title {
             vertical-align: top;
-            margin: -5px 0 0;
+            margin: 0 5px 0 0;
+            display: none;
```
Hmm I checked it to be redundant and removed it. Also I think you were mentioning about #24538 that removed saved/saving notifications from stream edit.

m-e-l-u-h-a-n commented:
I've updated this PR with the review suggestions, and fixed the stream symbol to be on same line as name.
![image](https://user-images.githubusercontent.com/63504956/227334702-76080438-a11e-42d6-808e-346871752cfb.png)

alya commented:
Thanks! When I change a stream or group name to a long one, the Saving/Saved notice doesn't fit inside its rectangle.

Looks good to me otherwise.

sahil839 commented:
Code changes look good to me.

> When I change a stream or group name to a long one, the Saving/Saved notice doesn't fit inside its rectangle.

We are anyways removing the notice in #24538.

sahil839 commented:
@m-e-l-u-h-a-n FYI, can you try this - https://github.com/zulip/zulip/pull/24788#issuecomment-1478391511

m-e-l-u-h-a-n commented:
> @m-e-l-u-h-a-n FYI, can you try this - [#24788 (comment)](https://github.com/zulip/zulip/pull/24788#issuecomment-1478391511)

@sahil839 I've updated this pr by adding word-break property to `.modal__title` .

> When I change a stream or group name to a long one, the Saving/Saved notice doesn't fit inside its rectangle.

Haven't made updates regarding this as these indicators are going to be removed in #24538, should I make updates regarding this as well?

sahil839 commented:
The stream icon and stream name are on separate lines in `Archive stream` modal. I am not sure whether it is because of this PR but would be good to fix.
![Screenshot from 2023-04-05 13-11-58](https://user-images.githubusercontent.com/35494118/230014598-9b9e9b86-36dc-4718-b8b5-ef34918fda21.png)

sahil839 commented:
Looks good to me. 

> Haven't made updates regarding this as these indicators are going to be removed in https://github.com/zulip/zulip/pull/24538, should I make updates regarding this as well?

#24538 is close to being completed, so we can skip it in this PR. I have asked the author of the PR to rebase the PR.

timabbott commented on web/src/user_groups_settings_ui.js:
```diff
@@ -19,6 +19,11 @@ import * as user_groups from "./user_groups";
+// Ideally this should be included in page params.
+// Like we have page_params.max_stream_name_length` and
+// `page_params.max_stream_description_length` for streams.
+export const max_user_group_name_length = 100;
```
@laurynmm can I add doing this API tweak to your TODO list? It's not urgent but also probably quick, so probably best do wait until we have a moment with fewer big API changes about to merge.

timabbott commented:
Merged, thanks @m-e-l-u-h-a-n!
